### PR TITLE
[Merged by Bors] - feat: construct homomorphism from `MulSemiringAction`

### PR DESCRIPTION
**Pull request by tb65536**
This PR adds some missing API for constructing a homomorphism from a `MulSemiringAction`s. We already have `MulSemiringAction.toRingAut`.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Algebra/Algebra/Equiv.lean [hunk 1]
```diff
@@ -787,7 +787,20 @@ variable {M G : Type*} (R A : Type*) [CommSemiring R] [Semiring A] [Algebra R A]
 
 section
 
-variable [Group G] [MulSemiringAction G A] [SMulCommClass G R A]
+variable [Group G]
+
+/-- Construct a `MulSemiringAction` from a homomorphism `G →* A ≃ₐ[R] A`. -/
```
Comment (Ruben-VandeVelde) [inline Mathlib/Algebra/Algebra/Equiv.lean:792]: I'm not sure what the precedence is between `→*` and `≃ₐ[R]`. Can you parenthesise the comment, at least?

File: Mathlib/Algebra/Algebra/Equiv.lean [hunk 2]
```diff
@@ -787,7 +787,20 @@ variable {M G : Type*} (R A : Type*) [CommSemiring R] [Semiring A] [Algebra R A]
...
+@[simps]
+def ofAlgEquivHom (h : G →* A ≃ₐ[R] A) : MulSemiringAction G A where
+  smul := fun g r ↦ h g r
```
Comment (Ruben-VandeVelde) [inline Mathlib/Algebra/Algebra/Equiv.lean:795]:
```suggestion
  smul g r := h g r
```
Also below

**Review by Ruben-VandeVelde — APPROVED** [review]

**Review by Ruben-VandeVelde — APPROVED** [review]
maintainer merge

Thanks!

File: Mathlib/Algebra/Algebra/Equiv.lean [hunk 3]
```diff
@@ -787,7 +787,20 @@ variable {M G : Type*} (R A : Type*) [CommSemiring R] [Semiring A] [Algebra R A]
...
+/-- Construct a `MulSemiringAction` from a homomorphism `G →* (A ≃ₐ[R] A)`. -/
+@[simps]
+def ofAlgEquivHom (h : G →* (A ≃ₐ[R] A)) : MulSemiringAction G A where
+  smul g r := h g r
+  one_smul := DFunLike.ext_iff.mp (map_one h)
+  mul_smul g g' := DFunLike.ext_iff.mp (map_mul h g g')
+  smul_zero g := map_zero (h g)
+  smul_add g := map_add (h g)
+  smul_one g := map_one (h g)
+  smul_mul g := map_mul (h g)
```
Comment (eric-wieser) [inline Mathlib/Algebra/Algebra/Equiv.lean:801]: I think this is just `MulSemiringAction.compHom`?

File: Mathlib/Algebra/Ring/Aut.lean [hunk 4]
```diff
@@ -96,6 +96,17 @@ instance apply_faithfulSMul : FaithfulSMul (RingAut R) R :=
...
+/-- Construct a `MulSemiringAction` from a homomorphism `G →* RingAut R`. -/
+@[simps]
+def _root_.MulSemiringAction.ofRingAut (h : G →* RingAut R) : MulSemiringAction G R where
+  smul g r := h g r
+  one_smul := DFunLike.ext_iff.mp (map_one h)
+  mul_smul g g' := DFunLike.ext_iff.mp (map_mul h g g')
+  smul_zero g := map_zero (h g)
+  smul_add g := map_add (h g)
+  smul_one g := map_one (h g)
+  smul_mul g := map_mul (h g)
```
Comment (eric-wieser) [inline Mathlib/Algebra/Ring/Aut.lean:108]: This one should also just be `MulSemiringAction.compHom`.

**Review by eric-wieser — COMMENTED** [review]
FWIW, I have the following table in [my thesis](https://www.repository.cam.ac.uk/items/9ad1ac65-48c4-49cd-a00c-3cec6397a93b) about this and the use of `compHom` (warning: lean 3!)

image

File: Mathlib/Algebra/Algebra/Equiv.lean [hunk 5]
```diff
@@ -801,6 +814,13 @@ theorem toAlgEquiv_injective [FaithfulSMul G A] :
     Function.Injective (MulSemiringAction.toAlgEquiv R A : G → A ≃ₐ[R] A) := fun _ _ h =>
   eq_of_smul_eq_smul fun r => AlgEquiv.ext_iff.1 h r
 
+/-- Each element of the group defines an algebra equivalence. -/
+@[simps]
+def toAlgEquivHom : G →* A ≃ₐ[R] A where
```
[the remark this answers is not in the document]
Comment (eric-wieser) [reply Mathlib/Algebra/Algebra/Equiv.lean:819]: In fact, I think this should be called `toAlgAut` and reference `MulSemiringAction.toRingAut` and `DistribMulAction.toModuleEnd`.

**Review by eric-wieser — APPROVED** [review]
bors merge

Thanks!

**Pull request closed without merging**